src/trips/schemas.py

```diff
@@ -104,3 +104,21 @@ class SRouteOutput(BaseModel):
     destination_id: int
 
     model_config = ConfigDict(from_attributes=True)
+
+
+class SHighlightInput(BaseModel):
+    """Create new highlight."""
+
+    name: str
+    description: str
+    route_id: int | None
+    location_id: int | None
```

Comment:
Раз ты поставил правило на уровне БД, что только одно из двух полей может быть заполнено, то давай и для пользователей API тут проверку поставим.

```suggestion
class SHighlightInput(BaseModel):
    """Create new highlight."""

    name: str
    description: str
    route_id: int | None
    location_id: int | None
    
    @model_validator(mode="before")
    @classmethod
    def check_one_parent_present(cls, data: Any):
        route_id, location_id = data.get('route_id'), data.get('location_id')
        if any([route_id is None and location_id is None, route_id is not None and location_id is not None]):
            raise ValueError("Either route_id or location_id must be set, but not both.")
        return data
    
```